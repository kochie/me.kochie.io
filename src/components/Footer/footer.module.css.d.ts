export const footer: string;
export const guitar: string;
export const jello: string;
