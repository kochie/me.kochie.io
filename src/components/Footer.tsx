import * as React from "react";

import '../styles/Footer.scss'

export default class Footer extends React.Component {
  render() {
    return <div className="footer"/>;
  }
}
