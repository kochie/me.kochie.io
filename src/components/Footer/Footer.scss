.footer {
  background-color: darkgrey;
  grid-area: footer;
  height: 75px;

  li::before {
    border-left: 2px solid #1f2326;
  }

  li {
    font-family: "Source Sans Pro", Helvetica, sans-serif;
    float: left;
  }

  ul {
    list-style: none;
  }
}
