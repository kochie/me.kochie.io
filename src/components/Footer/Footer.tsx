import React, { useRef } from "react";

import { footer, guitar, jello } from "./footer.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { findIconDefinition } from "@fortawesome/fontawesome-svg-core";

import "animate.css"

export default () => {
  const playSound = () => {
    const g = document.getElementById('guitar') 
    g.classList.add(jello)
    g.addEventListener('animationend', () => {
      g.classList.remove(jello)
    })

    const sound = new Audio("/audio/AcousticPopMelody.mp3")
    sound.play()
  }
  const updateTime = process.env.buildTime;
  return (
    <nav className={footer}>
      <ul>
        <li>
          <FontAwesomeIcon
            style={{paddingRight: '5px'}}
            icon={findIconDefinition({ prefix: "fad", iconName: "copyright" })}
          />
          Robert Koch</li>
        <li id="guitar">
          <FontAwesomeIcon
          onClick={playSound}
          className={guitar}
            icon={findIconDefinition({ prefix: "fad", iconName: "guitar" })}
          />
        </li>
        <li>
          <span>
            Updated on <time dateTime={updateTime}>{updateTime}</time>
          </span>
        </li>
      </ul>
    </nav>
  );
};
