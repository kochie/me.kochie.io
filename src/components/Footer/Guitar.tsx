'use client'

import { faGuitar } from '@fortawesome/pro-duotone-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import style from './footer.module.css'

export const Guitar = () => {
  const playSound = (): void => {
    const g = document.getElementById('guitar')
    g.classList.add(style.jello)
    g.addEventListener('animationend', () => {
      g.classList.remove(style.jello)
    })

    const sound = new Audio('/audio/AcousticPopMelody.mp3')
    sound.play()
  }

  return <FontAwesomeIcon onClick={playSound} icon={faGuitar} size="2x" />
}
