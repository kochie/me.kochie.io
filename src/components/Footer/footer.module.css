.footer {
  background-color: darkgrey;
  grid-area: footer;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  padding: 30px;
  font-family: 'Source Sans Pro', Helvetica, sans-serif;
  align-items: center;
}

.name {
  text-align: start;
}
.guitar {
  text-align: center;
}
.update {
  text-align: end;
}

@keyframes jello {
  from,
  11.1%,
  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

  22.2% {
    -webkit-transform: skewX(-12.5deg) skewY(-12.5deg);
    transform: skewX(-12.5deg) skewY(-12.5deg);
  }

  33.3% {
    -webkit-transform: skewX(6.25deg) skewY(6.25deg);
    transform: skewX(6.25deg) skewY(6.25deg);
  }

  44.4% {
    -webkit-transform: skewX(-3.125deg) skewY(-3.125deg);
    transform: skewX(-3.125deg) skewY(-3.125deg);
  }

  55.5% {
    -webkit-transform: skewX(1.5625deg) skewY(1.5625deg);
    transform: skewX(1.5625deg) skewY(1.5625deg);
  }

  66.6% {
    -webkit-transform: skewX(-0.78125deg) skewY(-0.78125deg);
    transform: skewX(-0.78125deg) skewY(-0.78125deg);
  }

  77.7% {
    -webkit-transform: skewX(0.390625deg) skewY(0.390625deg);
    transform: skewX(0.390625deg) skewY(0.390625deg);
  }

  88.8% {
    -webkit-transform: skewX(-0.1953125deg) skewY(-0.1953125deg);
    transform: skewX(-0.1953125deg) skewY(-0.1953125deg);
  }
}

.guitar svg {
  transition: all 0.5s;
  transform-origin: center;
}

.guitar svg:hover {
  --fa-primary-color: brown;
  --fa-secondary-color: gold;
  transform: scale(1.5);
}

.jello {
  animation-name: jello;
  animation-duration: 1s;
  /* animation-delay: 0.5s; */
}

@media (--mobile-size) {
  .footer ul {
    padding: 0 10px;
    font-size: small;
  }
}
