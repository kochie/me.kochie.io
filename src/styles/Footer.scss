.footer {
  background-color: darkgrey;
  grid-area: footer;
  height: 75px;
}