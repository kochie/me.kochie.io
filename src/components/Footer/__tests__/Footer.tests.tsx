import * as renderer from 'react-test-renderer'
import * as React from 'react'

import Footer from '../Footer'

it('footer renders correctly', () => {
  const tree = renderer.create(<Footer />).toJSON()
  expect(tree).toMatchSnapshot()
})
