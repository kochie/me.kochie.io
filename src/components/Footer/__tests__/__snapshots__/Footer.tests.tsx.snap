// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<nav
  className="footer"
>
  <ul>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-copyright fa-w-16 "
        data-icon="copyright"
        data-prefix="fad"
        focusable="false"
        role="img"
        style={
          Object {
            "paddingRight": "5px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-group"
          style={Object {}}
        >
          <path
            className="fa-secondary"
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm117.13 346.75c-1.59 1.87-39.77 45.73-109.85 45.73-84.69 0-144.48-63.26-144.48-145.56 0-81.31 62-143.4 143.76-143.4 67 0 102 37.31 103.42 38.9a12 12 0 0 1 1.24 14.58l-22.38 34.7a12 12 0 0 1-16.59 3.57 11.79 11.79 0 0 1-1.64-1.27c-.24-.21-26.53-23.88-61.88-23.88-46.12 0-73.92 33.58-73.92 76.08 0 39.61 25.52 79.7 74.28 79.7 38.7 0 65.28-28.34 65.54-28.63a12 12 0 0 1 16.95-.73h.05a12.2 12.2 0 0 1 1.55 1.74l24.55 33.58a12 12 0 0 1-.6 14.85z"
            fill="currentColor"
            style={Object {}}
          />
          <path
            className="fa-primary"
            d="M373.13 354.75c-1.59 1.87-39.77 45.73-109.85 45.73-84.69 0-144.48-63.26-144.48-145.56 0-81.31 62-143.4 143.76-143.4 67 0 102 37.31 103.42 38.9a12 12 0 0 1 1.24 14.58l-22.38 34.7a12 12 0 0 1-16.59 3.57 11.79 11.79 0 0 1-1.64-1.27c-.24-.21-26.53-23.88-61.88-23.88-46.12 0-73.92 33.58-73.92 76.08 0 39.61 25.52 79.7 74.28 79.7 38.7 0 65.28-28.34 65.54-28.63a12 12 0 0 1 16.95-.73h.05a12.2 12.2 0 0 1 1.55 1.74l24.55 33.58a12 12 0 0 1-.6 14.85z"
            fill="currentColor"
            style={Object {}}
          />
        </g>
      </svg>
      Robert Koch
    </li>
    <li
      id="guitar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-guitar fa-w-16 guitar"
        data-icon="guitar"
        data-prefix="fad"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-group"
          style={Object {}}
        >
          <path
            className="fa-secondary"
            d="M348.88 267.67c7.72 33.39.86 66.87-22 89.75a82.56 82.56 0 0 1-33.47 20.25c-18.54 6-32.67 23.29-34.43 42.15-2.28 23.8-11.48 45.82-28.45 62.7C185 528 103.08 519.94 47.57 464.45s-63.67-137.45-18-183c16.87-17 38.9-26.17 62.59-28.44 18.95-1.77 36.13-15.9 42.15-34.46a82.45 82.45 0 0 1 20.24-33.47c22.9-22.91 56.42-29.81 89.86-22 14.62 3.41 28 10.39 40.15 19.17l-74 74c-.85 0-1.65-.25-2.52-.25a48 48 0 1 0 48 48c0-.84-.21-1.62-.25-2.45l74-74.06c8.74 12.2 15.72 25.51 19.09 40.18z"
            fill="currentColor"
            style={Object {}}
          />
          <path
            className="fa-primary"
            d="M502.63 84.22l-46.32 46.32a35.26 35.26 0 0 1-13.79 8.53l-36.47 12.16-150.33 150.35a47.77 47.77 0 0 0-45.22-45.29L360.79 106 373 69.49a35.26 35.26 0 0 1 8.53-13.79l46.26-46.33a32 32 0 0 1 45.26 0L502.63 39a32 32 0 0 1 0 45.22z"
            fill="currentColor"
            style={Object {}}
          />
        </g>
      </svg>
    </li>
    <li>
      <span>
        Updated on 
        <time />
      </span>
    </li>
  </ul>
</nav>
`;
